cpp/include/cudf/io/text/trie.hpp

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/detail/utilities/vector_factories.hpp>
+#include <cudf/io/text/multistate.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+#include <queue>
+#include <string>
+#include <vector>
+
+namespace {
```

Comment:
Don't use anonymous namespaces in headers.

Reply:
Already deleted in a commit I'm waiting to push. :disappointed:

Reply:
I feel like this is the most common feedback you have to give me. :)